auth/src/krb.cpp

```diff
@@ -0,0 +1,942 @@
+                        THROW(err.code(), err.result());
+                    }
+                }
+
+                gss_release_buffer(&minorStatus, &send_tok);
+
+                if (majorStatus == GSS_S_CONTINUE_NEEDED) {
+                    recv_tok.value = &ikrbScratchBuffer;
+                    recv_tok.length = SCRATCH_BUFFER_SIZE;
+                    unsigned int bytes_read;
+                    if (const auto ret = krb_receive_token(fd, &recv_tok, &bytes_read); !ret.ok()) {
+                        gss_release_name( &minorStatus, &target_name );
+                        const auto err = PASSMSG("Error reading KRB token.", ret);
+                        THROW(err.code(), err.result());
+                    }
+
+                    tokenPtr = &recv_tok;
+                }
+            }
+            while (majorStatus == GSS_S_CONTINUE_NEEDED);
+
+            resp["next_operation"] = AUTH_CLIENT_AUTH_RESPONSE;
+
+            return resp;
+        } // kerberos_auth_establish_context
+
+        json kerberos_auth_client_response(rcComm_t& comm, const json& req)
+        {
+            namespace ie = irods::experimental;
+
+            if (auto [e, n] = verify(req, {"user_name","zone_name"}); !e) {
+                THROW(SYS_INVALID_INPUT_PARAM,
+                      fmt::format("[{}:{}] missing [{}] in request", __func__, __LINE__, n));
+            }
+
+            json svr_req{req};
+            svr_req["next_operation"] = AUTH_AGENT_AUTH_RESPONSE;
+            auto resp = request(comm, svr_req);
+
+            comm.loggedIn = 1;
+
+            resp["next_operation"] = ie::auth::flow_complete;
+
+            return resp;
+        } // kerberos_auth_client_response
+
+#ifdef RODS_SERVER
+        json kerberos_auth_agent_request(rsComm_t& comm, const json& req)
+        {
+            if (krbAuthReqStatus == 1) {
+                krbAuthReqStatus = 0;
+                if (krbAuthReqError != 0) {
+                    rodsLogAndErrorMsg(LOG_NOTICE, &comm.rError, krbAuthReqError, krbAuthReqErrorMsg);
+                    THROW(krbAuthReqError, "A KRB auth request error has occurred.");
+                }
+            }
+
+            const auto kerberos_name = irods::get_server_property<std::string>("KerberosServicePrincipal");
+
+            constexpr bool client = false;
+            const auto service_name = krb_setup_creds(&comm.rError, client, kerberos_name);
+
+            comm.gsiRequest = 1;
+            if (comm.auth_scheme) {
+                free(comm.auth_scheme);
+            }
+            comm.auth_scheme = strdup(irods::AUTH_KRB_SCHEME.c_str());
+
+            json resp{req};
+            resp["request_result"] = service_name;
+            return resp;
+        } // kerberos_auth_agent_request
+
+        json kerberos_auth_agent_response(rsComm_t& comm, const json& req)
+        {
+            if (auto [e, n] = verify(req, {"zone_name", "user_name"}); !e) {
```

Comment:
We should give names to `zone_name` and `user_name` since they are referenced in different places across this file.

Reply:
Perhaps something can be added the dev package for this as these are common across all of the auth plugins. Will make an issue

Reply:
I have added this to the list of tasks for wrapping up #5479: https://github.com/irods/irods/issues/5479#issuecomment-1115234800

Reply:
This is not done yet, but will be part of the final sweep across the plugins in the above-linked issue. Marking resolved.